lib/mongo.js

```diff
@@ -0,0 +1,77 @@
+const { MongoClient, ObjectId } = require('mongodb')
+
+const MONGO_DB = 'flossbank_db'
+const MAINTAINERS_COLLECTION = 'users'
+
+class Mongo {
+  constructor ({ config, log }) {
+    this.log = log
+    this.config = config
+    this.db = null
+    this.mongoClient = null
+  }
+
+  async connect () {
+    const mongoUri = await this.config.getMongoUri()
+    this.mongoClient = new MongoClient(mongoUri, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    })
+    await this.mongoClient.connect()
+
+    this.db = this.mongoClient.db(MONGO_DB)
+  }
+
+  async close () {
+    if (this.mongoClient) return this.mongoClient.close()
+  }
+
+  async getPendingMaintainerPayouts ({ maintainerId }) {
+    // Grab all maintainers who have a payout method (ILP pointer or otherwise)
+    // and also have payouts that have not been paid
+    const aggregationPipeline = [
+      {
+        $match: {
+          _id: new ObjectId(maintainerId),
+          'payoutInfo.ilpPointer': {
```

Comment:
realizing this is not as effective as just checking for existence of the pointer in process.js and bailing if it isn't there.

Reply:
i dont understand, can you explain?

Reply:
nvm now I see that ur checking for !maintainer in process, and this would return an empty array if the maintainer has no pointer, so it should all work out 👍